### ISSUE TITLE
PS 7.3.x is using a prerelease version of System.Security.AccessControl

### ISSUE THREAD

**Issue by sdwheeler** [PowerShell/PowerShell#19930]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

This is causing problems when we try to build the .NET API docs because you can load a pre-release dependency in a release product. This is only affecting publishing for 7.3. We don't have this problem with 7.2.x or 7.4.x.

I talked with @adityapatwardhan about this. He said it was a backport regression in 7.3.

### Expected behavior

```console
See above.
```

Affected code is can be found here: https://github.com/search?q=repo%3APowerShell%2FPowerShell%206.0.2-mauipre.1.22102.15&type=code

### Actual behavior

```console
n/a
```

### Error details

```console
The error from Nuget.exe for is:
"Unable to find a version of 'System.Security.AccessControl' that is compatible with 'System.Management.Automation 7.3.5 constraint: System.Security.AccessControl (>= 6.0.2-mauipre.1.22102.15)'."
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.5
PSEdition                      Core
GitCommitId                    7.3.5
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iSazonov** [comment]
Fixed by #19931

**Comment by zitmen** [comment]
Hi,

the same issue leaked to the 7.4.0
![image](https://github.com/PowerShell/PowerShell/assets/206394/e972a247-41fa-4386-a579-aab684eb6644)

Could I ask for a fix?

Thanks!
